scripts/runChainTests.ts

```diff
@@ -0,0 +1,195 @@
+
+	// Set the ws url env var
+	setWsUrl(wsUrl);
+
+	console.log('Launching Sidecar...');
+	const sidecarStart = await launchProcess(SasStartOpts);
+
+	if (sidecarStart === Success) {
+		// Sidecar successfully launched, and jest will now get called
+		console.log('Launching jest...');
+		const jest = await launchProcess(JestProcOpts);
+
+		if (jest === Success) {
+			killAll();
+			return true;
+		} else {
+			killAll();
+			return false;
+		}
+	} else {
+		console.error('Error launching sidecar... exiting...');
+		killAll();
+		process.exit(1);
+	}
+};
+
+// Kill all processes spawned and tracked by this file.
+const killAll = () => {
+	console.log('Killing all processes...');
+	for (const key of Object.keys(procs)) {
+		if (!procs[key].killed) {
+			try {
+				console.log(`Killing ${key}`);
+				// Kill child and all its descendants.
+				process.kill(-procs[key].pid, 'SIGTERM');
+				process.kill(-procs[key].pid, 'SIGKILL');
+			} catch (e) {
+				/**
+				 * The error we are catching here silently, is when `-procs[key].pid` takes
+				 * the range of all pid's inside of the subprocess group created with
+				 * `spawn`, and one of the process's is either already closed or doesn't exist anymore.
+				 *
+				 * ex: `Error: kill ESRCH`
+				 *
+				 * This is a very specific use case of an empty catch block and is used
+				 * outside of the scope of the API therefore justifiable, and should be used cautiously
+				 * elsewhere.
+				 */
+			}
+		}
+	}
+};
+
+const checkTests = (...args: boolean[]) => {
+	const testStatus = args.every((test) => test);
+
+	if (testStatus) {
+		console.log('[PASSED] All Tests Passed!');
+		process.exit(0);
+	} else {
+		console.log('[FAILED] Some Tests Failed!');
+		process.exit(1);
+	}
+};
+
+const main = async (args: Namespace): Promise<void> => {
+	const { Failed } = StatusCode;
+
+	// Build sidecar
+	console.log('Building Sidecar...');
+	const sidecarBuild = await launchProcess(defaultSasBuildOpts);
+
+	// When sidecar fails to build, we kill all process's and exit
+	if (sidecarBuild === Failed) {
+		console.error('Sidecar failed to build, exiting...');
+		// Kill all processes
+		killAll();
+		// Exit program
+		process.exit();
+	}
+
+	if (args.chain) {
+		const selectedChain = await launchChainTest(args.chain);
+
+		checkTests(selectedChain);
+	} else {
+		// Test the e2e tests against polkadot
+		const polkadotTest = await launchChainTest('polkadot');
+
+		// Test the e2e tests against kusama
+		const kusamaTest = await launchChainTest('kusama');
+
+		// Test the e2e tests against westend
+		const westendTest = await launchChainTest('westend');
+
+		checkTests(polkadotTest, kusamaTest, westendTest);
+	}
+};
+
+/**
+ * Arg Parser
+ */
+const parser = new ArgumentParser();
+
+parser.add_argument('--chain', {
+	choices: ['polkadot', 'kusama', 'westend'],
+});
+
+const args = parser.parse_args() as Namespace;
+
+/**
+ * Signal interrupt
+ */
+process.on('SIGINT', function () {
```

Comment:
I wonder whether iot's worth covering a few more bases and also cleaning up when SIGKILL, SIGTERM (eg when kill command used) and SIGHUP (eg when terminal window closed) are sent?

Reply:
Yea I agree, I just added SIGHUP. 

> cleaning up when SIGKILL, SIGTERM (eg when kill command used)

Curious what you mean by that?

Reply:
Oh, I just meant that SIGTERM is the default signal sent when `kill` is used. Actually, I don't think you can catch SIGKILL looking at the `kill` man page so that's not useful anyway!

Reply:
(I thnk that catching the result of ctrl+c is most important though and you've done that, and now you've added the "terminal" closed signal that seems good to me!)

Reply:
Ahh okay cool i see what you are referencing at :)